### DISCUSSION TITLE
Schema migrations and reflect_table

### DISCUSSION THREAD

**Discussion by lukasz-madon** [sqlalchemy/alembic#1407] [Usage Questions]
There are a few examples of migrations in the documentation and tutorials. There seems to be two ways of doing migrations

1. Explicit SQL
2. ORM via `table()`

```
   users = table('users',
        column('first_name', String),
        column('last_name', String),
        column('full_name', String)
    )
    conn = op.get_bind()
    conn.execute(
        users.update().values(
            full_name=sa.func.concat(users.c.first_name, ' ', users.c.last_name)
        )
    )
```
There 3rd option would be to use `reflect_table`

```
users = op.reflect_table('users')
users.update().values(
```

Is reflection OK to use in this case? What are the pros and cons? Can `reflect_table` bit me with without explicit typing?

**Comment by CaselIT** [comment]
Hi,

Well the main drawback of reflect table is that you need to have a db available, meaning you can't use alembic in offline mode, ie to generate sql files to apply to the db by some other mean.

Other than that I would think it should be safe if you use only alembic to modify the schema of the db

**Comment by lukasz-madon** [comment]
I guess you also lose typechecking - reflection can make conversion implicit.

**Reply by CaselIT** [reply]
In a way yes, but at the moment `table()` or `Table` don't offer much type checking, at least in python using mypy and similar tools.
If you meant more the sql type specified into `table` etc, then yes when reflecting a table you get the type in the db
